### PR TITLE
Add Github CI

### PR DESCRIPTION
**Pull request by devmotion**

### CODE REVIEW

File: .github/workflows/CompatHelper.yml [hunk 1]
```diff
@@ -14,4 +12,5 @@ jobs:
       - name: CompatHelper.main()
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          COMPATHELPER_PRIV: ${{ secrets.COMPATHELPER_PRIV }}
```
Comment (devmotion) [inline .github/workflows/CompatHelper.yml:15]: This secret has to be added: https://github.com/JuliaRegistries/CompatHelper.jl#12-set-up-the-ssh-deploy-key-optional

Comment (cpfiffer) [reply .github/workflows/CompatHelper.yml:15]: I think the trend here is towards using Github for all our repos, so we should probably just use an organization-wide secret.

@yebai can you make me a TuringLang admin for like a day or something so I can set up the keys? It should only take me like 5 minutes total.

Comment (devmotion) [reply .github/workflows/CompatHelper.yml:15]: Yes, that's probably even better 👍

Comment (cpfiffer) [reply .github/workflows/CompatHelper.yml:15]: I added a repo-specific token for the moment, so it should work.

Comment (devmotion) [reply .github/workflows/CompatHelper.yml:15]: Can you approve if you think that the PR is ready to be merged? Then we could run the tests in the other PR.

**Review by cpfiffer — APPROVED** [review]

**Pull request merged**